### PR TITLE
Regenearte Cognitive Services Mgmt. SDK with latest Spec and AutoRest (targeting release as 2.1)

### PR DESCRIPTION
**Pull request by felixwa**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj [hunk 1]
```diff
@@ -6,18 +6,26 @@
...
 		<AssemblyName>Microsoft.Azure.Management.CognitiveServices</AssemblyName>
 		<PackageTags>Microsoft Azure Cognitive Services management;Cognitive Services;Cognitive Services management</PackageTags>
-		<PackageReleaseNotes/>
+		<PackageReleaseNotes />
 	</PropertyGroup>
 	<PropertyGroup>
 		<TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
 	</PropertyGroup>
+	<ItemGroup>
+	  <Compile Remove="Properties\AssemblyInfo.cs" />
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj]: @felixwa please remove from line 17 to 19
also remove line 24 to 27

File: src/SDKs/CognitiveServices/management/CognitiveServices.Tests/CognitiveServices.Tests.csproj [hunk 2]
```diff
@@ -10,6 +10,9 @@
   <PropertyGroup>
     <TargetFrameworks>netcoreapp1.1</TargetFrameworks>
   </PropertyGroup>
+  <ItemGroup>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/management/CognitiveServices.Tests/CognitiveServices.Tests.csproj]: @felixwa please remove line 13 to 15 and line 35 to 38

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Please link the spec PR.
Also use generate.cmd to generated sdk, this will create a meta data file (.txt)

File: src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj (hunk 1 above)
Comment (felixwa) [reply src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj]: @shahabhijeet , I'm seeing CS0579 if AssemblyInfo.cs is included.
obj\Debug\netstandard1.4\Microsoft.Azure.Management.CognitiveServices.AssemblyInfo.cs(14,12,14,54): error CS0579: Duplicate 'System.Reflection.AssemblyCompanyAttribute' attribute

Can you please tell me how to mitigate this error? Should I add below?
  <PropertyGroup>
    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
  </PropertyGroup>

**Comment by felixwa** [comment]
Spec PR: https://github.com/Azure/azure-rest-api-specs/pull/1981

**Comment by felixwa** [comment]
The changes were generated generate.cmd. I also fixed an folder structure issue with generate.cmd in this PR.
A remaining issue with generate.cmd is that now it generates the folder under SDKs\CognitiveServices folder but it should be SDKs\CognitiveServices\management folder. @shahabhijeet if you know how to fix this I can also do that in this PR.

File: src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj (hunk 1 above)
Comment (felixwa) [reply src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj]: I talked to @DavidLiCIG  and he said we can remove this AssemblyInfo.cs file so I removed it.

File: src/SDKs/CognitiveServices/management/CognitiveServices.Tests/CognitiveServices.Tests.csproj (hunk 2 above)
Comment (felixwa) [reply src/SDKs/CognitiveServices/management/CognitiveServices.Tests/CognitiveServices.Tests.csproj]: Remove and also removed AssemblyInfo.cs

File: src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj [hunk 3]
```diff
@@ -6,17 +6,21 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.CognitiveServices</PackageId>
 		<Description>Microsoft Azure Management Cognitive Services Library</Description>
-		<Version>2.0.0</Version>
+		<Version>2.1.0</Version>
 		<AssemblyName>Microsoft.Azure.Management.CognitiveServices</AssemblyName>
 		<PackageTags>Microsoft Azure Cognitive Services management;Cognitive Services;Cognitive Services management</PackageTags>
-		<PackageReleaseNotes/>
+		<PackageReleaseNotes />
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj]: Can you add package release notes to specify what is new in this release.

File: src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj [hunk 4]
```diff
@@ -6,17 +6,21 @@
...
 	</PropertyGroup>
 	<PropertyGroup>
 		<TargetFrameworks>net452;netstandard1.4</TargetFrameworks>
 	</PropertyGroup>
 	<ItemGroup>
 		<Content Include="notice.txt" />
 	</ItemGroup>
+	
+	<ItemGroup>
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/management/Management.CognitiveServices/Microsoft.Azure.Management.CognitiveServices.csproj]: @felixwa please remove this item group as it's not needed to specify explicitly about directory structure.
Makes the project xml unnecessarily large.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Missing meta data file when generated using generate.cmd.
Any reason AssemblyInfo.cs was deleted.

File: src/SDKs/_metadata/cognitiveservices_resource-manager.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+2017-12-07 03:31:20 UTC
+
+1) azure-rest-api-specs repository information
+GitHub user: felixwa
```
Comment (felixwa) [inline src/SDKs/_metadata/cognitiveservices_resource-manager.txt]: @shahabhijeet , I know this is generated by my own fork. Please take a look at this PR:https://github.com/Azure/azure-rest-api-specs/pull/2103. After that PR is approved and merged I will regenerate.

**Comment by felixwa** [comment]
@shahabhijeet , the reason I removed AssemblyInfo.cs is because it generates compiler error CS0579

obj\Debug\netstandard1.4\Microsoft.Azure.Management.CognitiveServices.AssemblyInfo.cs(14,12,14,54): error CS0579: Duplicate 'System.Reflection.AssemblyCompanyAttribute' attribute

**Comment by shahabhijeet** [comment]
@felixwa if you simply do
msbuild build.proj it will update your build tools and then if you try to rebuild it should not give you that error.
if it still does, let me know and I would like to see it and will fix it if required. But we cannot ship sdk without AssemblyInfo

**Comment by felixwa** [comment]
@shahabhijeet , now AssemblyInfo.cs is back and conflict is merged. Continue your review.

File: src/SDKs/CognitiveServices/management/Management.CognitiveServices/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Cognitive Services Management Library")]
 [assembly: AssemblyDescription("Provides Microsoft Azure Cognitive Services management functions for managing Microsoft Azure Cognitive Services accounts.")]
 
-[assembly: AssemblyVersion("2.0.0.0")]
-[assembly: AssemblyFileVersion("2.0.0.0")]
+[assembly: AssemblyVersion("2.1.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/CognitiveServices/management/Management.CognitiveServices/Properties/AssemblyInfo.cs]: @felixwa please do not update AssemblyVersion when you change the minor version of AssemblyFileVersion.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
The Spec PR is merged. Please do the following:
1) Pull latest from upstream
2) Regenerate your SDK
3) Msbuild build.proj to get the latest build tools
4) Build you sdk from command line or IDE
5) Create/update this PR.

File: src/SDKs/CognitiveServices/management/Management.CognitiveServices/Properties/AssemblyInfo.cs (hunk 6 above)
Comment (felixwa) [reply src/SDKs/CognitiveServices/management/Management.CognitiveServices/Properties/AssemblyInfo.cs]: @shahabhijeet , please note there is a breaking change here (provisionState is changed to string from enum), why do not increase AssemblyVersion in this case?

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**